airflow/models.py

```diff
@@ -311,22 +312,37 @@ class Connection(Base):
     schema = Column(String(500))
     login = Column(String(500))
     _password = Column('password', String(500))
-    is_encrypted = Column(Boolean, unique=False, default=False)
     port = Column(Integer())
+    is_encrypted = Column(Boolean, unique=False, default=False)
     extra = Column(String(5000))
 
     def __init__(
             self, conn_id=None, conn_type=None,
             host=None, login=None, password=None,
-            schema=None, port=None, extra=None):
+            schema=None, port=None, extra=None,
+            uri=None):
         self.conn_id = conn_id
         self.conn_type = conn_type
-        self.host = host
-        self.login = login
-        self.password = password
-        self.schema = schema
-        self.port = port
-        self.extra = extra
+        if uri:
+            self.parse_from_uri(uri)
+        else:
+            self.host = host
+            self.login = login
+            self.password = password
+            self.schema = schema
+            self.port = port
+            self.extra = extra
+
+    def parse_from_uri(self, uri):
+        temp_uri = urlparse(uri)
+        hostname = temp_uri.hostname or ''
+        if '%2f' in hostname:
```

Comment:
I'm surprised that `urlparse` doens't urldecode... I guess hostname is not expected to have slashes in it...